main.js

```diff
@@ -1,1007 +1,1002 @@
+        }
+        var longVal = Long.fromString(handle.toString('hex'), unsigned = true, radix = 16);
+
+        return longVal.toString();
+    }
+
+    function initPlaybackHandler(mpvPath) {
+
+        var playbackhandler = require('./playbackhandler/playbackhandler');
+        playbackhandler.initialize(getWindowId(playerWindow), mpvPath);
+        playbackhandler.registerMediaPlayerProtocol(electron.protocol, mainWindow);
+    }
+
+    setCommandLineSwitches();
+
+    var fullscreenOnShow = false;
+    var windowShowCount = 0;
+    function onWindowShow() {
+
+        windowShowCount++;
+        if (windowShowCount == 2) {
+
+            mainWindow.center();
+            mainWindow.focus();
+            initialShowEventsComplete = true;
+        }
+    }
+
+    app.on('quit', function () {
+        closeWindow(mainWindow);
+    });
+
+    function onPlayerWindowRestore() {
+        mainWindow.focus();
+    }
+
+    // This method will be called when Electron has finished
+    // initialization and is ready to create browser windows.
+    app.on('ready', function () {
+
+        var isWindows = require('is-windows');
+        var windowStatePath = getWindowStateDataPath();
+
+        var previousWindowInfo;
+        try {
+            previousWindowInfo = JSON.parse(require("fs").readFileSync(windowStatePath, 'utf8'));
+        }
+        catch (e) {
+            previousWindowInfo = {};
+        }
+
+        var windowOptions = {
+            transparent: false, //supportsTransparency,
+            frame: false,
+            resizable: false,
+            title: 'Jellyfin Theater',
+            minWidth: 1280,
+            minHeight: 720,
+            //alwaysOnTop: true,
+            skipTaskbar: isWindows() ? false : true,
+
+            //show: false,
+            backgroundColor: '#00000000',
+            center: true,
+            show: false,
+            resizable: useTrueFullScreen,
+
+            webPreferences: {
+                webSecurity: false,
+                webgl: false,
+                nodeIntegration: false,
+                plugins: false,
+                webaudio: true,
+                java: false,
+                allowDisplayingInsecureContent: true,
+                allowRunningInsecureContent: true,
+                experimentalFeatures: false,
+                devTools: enableDevTools
+            },
+
+            icon: __dirname + '/icon.ico'
+        };
+
+        windowOptions.width = previousWindowInfo.width || 1280;
+        windowOptions.height = previousWindowInfo.height || 720;
+        if (previousWindowInfo.x != null && previousWindowInfo.y != null) {
+            windowOptions.x = previousWindowInfo.x;
+            windowOptions.y = previousWindowInfo.y;
+        }
+
+        playerWindow = new BrowserWindow(windowOptions);
+
+        windowOptions.backgroundColor = '#00000000';
+        windowOptions.parent = playerWindow;
+        windowOptions.transparent = true;
+        windowOptions.resi
```

Comment:
This should verify that URL is correct first, probably by invoking a "system info" API. Otherwise one typo and user might be lost forever trying to reset a wrongly input URL...